pkg/virtctl/usbredir/usbredir.go

```diff
@@ -0,0 +1,217 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2017, 2020 Red Hat, Inc.
+ *
+ */
+
+package usbredir
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/exec"
+	"os/signal"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/kubevirt/pkg/virtctl/templates"
+)
+
+const (
+	USBREDIRSERVER = "usbredirserver"
+)
+
+func NewCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "usbredir (vendor:product) (VMI)",
+		Short:   "Redirect a usb device to a virtual machine instance.",
+		Example: usage(),
+		Args:    templates.ExactArgs("usb", 2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c := usbredirCommand{clientConfig: clientConfig}
+			return c.Run(cmd, args)
+		},
+	}
+	cmd.SetUsageTemplate(templates.UsageTemplate())
+	return cmd
+}
+
+type usbredirCommand struct {
+	clientConfig clientcmd.ClientConfig
+}
+
+func (usbredirCmd *usbredirCommand) Run(command *cobra.Command, args []string) error {
+	namespace, _, err := usbredirCmd.clientConfig.Namespace()
+	if err != nil {
+		return err
+	}
+
+	virtCli, err := kubecli.GetKubevirtClientFromClientConfig(usbredirCmd.clientConfig)
+	if err != nil {
+		return err
+	}
+
+	vmiArg := args[1]
+	usbdeviceArg := args[0]
+
+	// Get connection to the websocket for usbredir subresource
+	usbredirVMI, err := virtCli.VirtualMachineInstance(namespace).USBRedir(vmiArg)
+	if err != nil {
+		return fmt.Errorf("Can't access VMI %s: %s", vmiArg, err.Error())
+	}
+
+	// We will connect the local USB device using usbredirserver's TCP server
+	// to the remote VM using the websocket.
+	pipeInReader, pipeInWriter := io.Pipe()
```

Comment:
Shouldn't we handle the close of writer?

Reply:
I've added defer _.Close() for both writers in their goroutines.